crates/examples/src/bin/dwarfdump.rs

```diff
@@ -1139,22 +1180,38 @@ fn dump_entries<R: Reader, W: Write>(
     w: &mut W,
     unit: gimli::UnitRef<R>,
     flags: &Flags,
+    desired_offset: Option<UnitOffset>,
 ) -> Result<()> {
     let mut spaces_buf = String::new();
     let mut deferred_macinfo = Vec::new();
     let mut deferred_macros = Vec::new();
 
     let mut entries = unit.entries_raw(None)?;
+    let mut found_desired_offset = desired_offset.is_none();
+    let mut baseline_depth = 0;
     while !entries.is_empty() {
         let offset = entries.next_offset();
         let depth = entries.next_depth();
         let abbrev = entries.read_abbreviation()?;
 
-        let mut indent = if depth >= 0 {
-            depth as usize * 2 + 2
+        let mut indent = if baseline_depth > 0 && baseline_depth >= depth {
+            break;
         } else {
-            2
+            (depth - baseline_depth) as usize * 2 + 2
```

Comment:
This can now be negative if `baseline_depth == 0 && depth < 0`, which is possible for malformed DWARF. It would be nice to keep the previous behaviour for this.

Reply:
Ah, I was wondering what that was for.